src/bugbug/data/bugbug_data/retriever.py

```diff
@@ -69,6 +70,13 @@ def retrieve_bugs(self):
         bug_ids = labels.get_all_bug_ids()
         bugzilla.download_bugs(bug_ids)
 
+        # Try to re-download inconsistent bugs, up to three times.
+        for i in range(3):
+            bug_ids = bug_snapshot.get_inconsistencies()
+            logger.info(f'Re-downloading {len(bug_ids)} bugs, as they were inconsistent')
+            bugzilla.delete_bugs(bug_ids)
+            bugzilla.download_bugs(bug_ids)
```

Comment:
Don't you want a stop condition here, so if the download is ok you don't retry un-necessary ?

Reply:
I'll do this. In the end the `bugzilla.delete_bugs` and `bugzilla.download_bugs` don't do anything if bug_ids is empty, but they needlessly read the bugs DB.